app/src/main/java/io/appium/settings/LocationService.java

```diff
@@ -172,45 +170,61 @@ public void run() {
         locationUpdatesTimer.schedule(locationUpdateTask, 0, UPDATE_INTERVAL_MS);
     }
 
-    private void updateMockLocationFactory(Intent intent) {
+    private Location getLocation(Intent intent, String providerName) {
         double longitude;
         try {
-            longitude = Double.valueOf(intent.getStringExtra("longitude"));
+            longitude = Double.parseDouble(intent.getStringExtra("longitude"));
         } catch (NumberFormatException e) {
             throw new IllegalArgumentException(
                     String.format("longitude should be a valid number. '%s' is given instead",
                             intent.getStringExtra(LONGITUDE_PARAMETER_KEY)));
         }
         double latitude;
         try {
-            latitude = Double.valueOf(intent.getStringExtra("latitude"));
+            latitude = Double.parseDouble(intent.getStringExtra("latitude"));
         } catch (NumberFormatException e) {
             throw new IllegalArgumentException(
                     String.format("latitude should be a valid number. '%s' is given instead",
                             intent.getStringExtra(LATITUDE_PARAMETER_KEY)));
         }
-        double altitude = 0.0;
+
+        double altitude = 0.0f;
```

Comment:
from android source code api 29:

    private double mLatitude = 0.0;
    private double mLongitude = 0.0;
    private double mAltitude = 0.0f;
    private float mSpeed = 0.0f;
    private float mBearing = 0.0f;
    private float mHorizontalAccuracyMeters = 0.0f;
    private float mVerticalAccuracyMeters = 0.0f;
    private float mSpeedAccuracyMetersPerSecond = 0.0f;
    private float mBearingAccuracyDegrees = 0.0f;

the mLatitude, mLongitude, mAltitude are type double

Reply:
> private double mAltitude = 0.0f;

I believe this might be a typo %)
Nevertheless such code is correct, because float values are assignable to variables of type double and are converted automatically